dnf/db/history.py

```diff
@@ -288,7 +289,10 @@ def swdb(self):
         """ Lazy initialize Swdb object """
         if not self._swdb:
             # _db_dir == persistdir which is prepended with installroot already
-            self._swdb = libdnf.transaction.Swdb(self.dbpath)
+            try:
+                self._swdb = libdnf.transaction.Swdb(self.dbpath)
+            except RuntimeError as ex:
+                raise DatabaseError(str(ex))
```

Comment:
I believe that the proper place for changing error type should be directly in swig. C++ error to Runtime error to DatabaseError is little bit too long. What about to impro it in DNF 5?